contracts/0.8.9/StakingRouter.sol

```diff
@@ -0,0 +1,570 @@
+        return module.lastDepositBlock;
+    }
+
+    function getStakingModuleActiveKeysCount(
+        uint24 _stakingModuleId
+    ) external view onlyRegisteredStakingModule(_stakingModuleId) returns (uint256) {
+        return IStakingModule(_getStakingModuleAddressById(_stakingModuleId)).getActiveKeysCount();
+    }
+
+    /**
+     * @dev calculate max count of depositable module keys based on the total prospective number of deposits
+     *
+     * @param _stakingModuleId id of the staking module to be deposited
+     * @param _totalDepositsCount total number of deposits to be made
+     * @return max depositable keys count
+     */
+    function getStakingModuleProspectiveMaxDepositableKeys(
+        uint24 _stakingModuleId,
+        uint256 _totalDepositsCount
+    ) external view onlyRegisteredStakingModule(_stakingModuleId) returns (uint256) {
+        return _getStakingModuleProspectiveMaxDepositableKeysByIndex(_getStakingModuleIndexById(_stakingModuleId), _totalDepositsCount);
+    }
+
+    /**
+     * @dev see {StakingRouter-getStakingModuleProspectiveMaxDepositableKeys}
+     *
+     * @param _stakingModuleIndex module index
+     * @param _totalDepositsCount total number of deposits to be made
+     * @return max depositable keys count
+     */
+    function _getStakingModuleProspectiveMaxDepositableKeysByIndex(
+        uint256 _stakingModuleIndex,
+        uint256 _totalDepositsCount
+    ) internal view returns (uint256) {
+        (, uint256[] memory newKeysAllocation, StakingModuleCache[] memory modulesCache) = _getKeysAllocation(_totalDepositsCount);
+        return newKeysAllocation[_stakingModuleIndex] - modulesCache[_stakingModuleIndex].activeKeysCount;
+    }
+
+    /**
+     * @notice return shares table
+     *
+     * @return recipients recipients list
+     * @return moduleFees fee of each recipient
+     * @return totalFee total fee to mint for each module and treasury
+     */
+    function getStakingRewardsDistribution()
+        external
+        view
+        returns (address[] memory recipients, uint16[] memory moduleFees, uint16 totalFee)
+    {
+        (uint256 totalActiveKeys, StakingModuleCache[] memory modulesCache) = _loadNotStoppedStakingModulesCache();
+        uint256 modulesCount = modulesCache.length;
+
+        /// @dev return empty response if there are no modules or active keys
+        if (modulesCount == 0 || totalActiveKeys == 0) {
+            return (new address[](0), new uint16[](0), 0);
+        }
+
+        recipients = new address[](modulesCount);
+        moduleFees = new uint16[](modulesCount);
+
+        uint256 rewardedModulesCount = 0;
+        uint256 moduleKeysShare;
+        for (uint256 i; i < modulesCount; ++i) {
+            /// @dev skip modules which have no active keys
+            if (modulesCache[i].activeKeysCount > 0) {
+                moduleKeysShare = ((modulesCache[i].activeKeysCount * TOTAL_BASIS_POINTS) / totalActiveKeys);
```

Comment:
Math would be broken in case when:
Curated module would have 144k validators
New modules wouldn't get rewards until 14 validatoros
Suggestion: increase accuracy

Reply:
fixed